### ISSUE TITLE
Bundler test suite hangs with simplecov

### ISSUE THREAD

**Issue by RochesterinNYC** [bundler/bundler#4192]
When adding simplecov to the test suite (through adding `require 'simplecov'` to `bin/rspec` and `require 'simplecov' and `SimpleCov.start` to `spec/spec_helper.rb), certain test cases hang indefinitely.

Hanging on `fetching dependencies with a mirrored source`:
```
$ bin/rspec spec/realworld/gemfile_source_header_spec.rb:4
Run options:
  include {:focused=>true, :locations=>{"./spec/realworld/gemfile_source_header_spec.rb"=>[4]}}
  exclude {:rubygems_master=>true, :rubygems=>"=< 2.5.0", :ruby=>"=< 2.2.3", :realworld=>true, :sudo=>true}

fetching dependencies with a mirrored source
/path/bundler/tmp/gems/base/gems/sinatra-1.2.7/lib/sinatra/base.rb:1258: warning: assigned but unused variable - e
.........Indefinite hanging here
```

**Comment by RochesterinNYC** [comment]
Regular output (without simplecov) for reference:
```
$ bin/rspec spec/realworld/gemfile_source_header_spec.rb:4
Run options:
  include {:focused=>true, :locations=>{"./spec/realworld/gemfile_source_header_spec.rb"=>[4]}}
  exclude {:rubygems_master=>true, :rubygems=>"=< 2.5.0", :ruby=>"=< 2.2.3", :realworld=>true, :sudo=>true}

fetching dependencies with a mirrored source
/path/bundler/tmp/gems/base/gems/sinatra-1.2.7/lib/sinatra/base.rb:1258: warning: assigned but unused variable - e
[2015-12-29 01:13:30] INFO  WEBrick 1.3.1
[2015-12-29 01:13:30] INFO  ruby 2.2.3 (2015-08-18) [x86_64-darwin14]
[2015-12-29 01:13:30] INFO  WEBrick::HTTPServer#start: pid=7268 port=21453
[2015-12-29 01:13:31] INFO  going to shutdown ...
[2015-12-29 01:13:32] INFO  WEBrick::HTTPServer#start done.
  sets the 'X-Gemfile-Source' header and bundles successfully


Retried examples: 0

Finished in 4.19 seconds (files took 0.4488 seconds to load)
1 example, 0 failures
```

**Comment by indirect** [comment]
@RochesterinNYC our tests do horrible, horrible things... most tests shell out to run `bundle` in a subshell, some tests monkeypatch net/http, some tests literally spin up webrick in a thread and then run `bundle` in a subshell to make requests against it. I would mostly expect that simplecov has no real way to keep track of all of that and produce useful output :/

We have very few unit tests (the ones we have today are in `spec/lib`), but it would be awesome to have simplecov coverage for the unit tests! Is there a way to have travis runs include simplecov and display the coverage results?

**Comment by RochesterinNYC** [comment]
Hm, yeah, after some experimentation, `simplecov` doesn't seem to play too well with `Popen` which we use for running a lot of the bundle commands in the test suite. Since the `Simplecov.start` is what starts tracking of code execution, then that bit would need to be put somewhere in the bundler app code (not really feasible) instead of the `spec_helper` if it were to track execution in all the subprocesses.

Similar issues @ colszowka/simplecov#346 and colszowka/simplecov#414

Using `simplecov` just for the unit tests might be viable. However, I don't see a `spec/lib` directory on the master branch? Is it on a different branch or something?

**Comment by indirect** [comment]
Ahh, sorry, I meant `spec/bundler`. Tests in that subdirectory map to the files inside `lib/bundler`.

**Comment by RochesterinNYC** [comment]
Essentially resolved by #4224. Code Climate's test coverage engine is basically a wrapper around simplecov.
